cpp/tests/join/mixed_join_tests.cu

```diff
@@ -778,6 +778,131 @@ TYPED_TEST(MixedLeftSemiJoinTest, BasicEquality)
              {1});
 }
 
+TYPED_TEST(MixedLeftSemiJoinTest, MixedLeftSemiJoinGatherMap)
+{
+  auto const col_ref_left_1  = cudf::ast::column_reference(0, cudf::ast::table_reference::LEFT);
+  auto const col_ref_right_1 = cudf::ast::column_reference(0, cudf::ast::table_reference::RIGHT);
+  auto left_one_greater_right_one =
+    cudf::ast::operation(cudf::ast::ast_operator::GREATER, col_ref_left_1, col_ref_right_1);
+
+  this->test({{2, 3, 9, 0, 1, 7, 4, 6, 5, 8}, {1, 2, 3, 4, 5, 6, 7, 8, 9, 0}},
+             {{6, 5, 9, 8, 10, 32}, {0, 1, 2, 3, 4, 5}, {7, 8, 9, 0, 1, 2}},
+             {0},
+             {1},
+             left_one_greater_right_one,
+             {2, 7, 8});
+}
+
+TYPED_TEST(MixedLeftSemiJoinTest, MixedLeftSemiJoinGatherMapLarge)
```

Comment:
New gtest that reproduces the bug that Spark reported.

Reply:
To test the scenario not covered by the current implementation, we need an input where the number of rows exceeds the block size. Given this condition, the current test seems to be overly complex. Is there a way we can simplify the test a bit?

Reply:
Separating the boilerplate of random data and nullmask generation, the test simply compares the number of rows in the output from `mixed_semi_join` and `conditional_left_semi_join` with an identical filter. I was actually thinking of comparing the returned columns themselves as well 😅

Reply:
Or I can modify the test at L781 to use larger input tables and use a ground truth expected vector.

Reply:
https://github.com/rapidsai/cudf/blob/dc57c1b1284816d0e5ed7493e6b661590c305511/cpp/tests/join/mixed_join_tests.cu#L64-L125

could we reuse those random data generators for this test case?

Reply:
Oh yes, I didn't even see this hence all this effort. Let me use these instead

Reply:
All done!

Reply:
@mhaseeb123 Thanks for your quick turnaround on the comments!